blocks/query-panel/category-select.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * External Dependencies
+ */
+import { flowRight } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { withAPIData } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { buildTermsTree } from '../../editor/utils/terms';
+import TermTreeSelect from '../term-tree-select';
+
+function CategorySelect( { label, noOptionLabel, categories, selectedCategory, onChange } ) {
+	return (
+		<TermTreeSelect
+			{ ...{ label, noOptionLabel } }
+			termsTree={ buildTermsTree( categories.data ) }
+			selectedTerm={ selectedCategory }
+			onChange={ onChange }
+		/>
+	);
+}
+
+const applyWithAPIData = withAPIData( () => ( {
+	categories: '/wp/v2/categories',
+} ) );
+
+export default flowRight( [
```

Comment:
The `flowRight` here is unnecessary at this point (with a single higher-order component) and could be written as:

```js
export default applyWithAPIData( CategorySelect );
```